DOCUMENTATION.md

```diff
@@ -2180,11 +2188,266 @@ const value = await map.get('key');
 console.log(value);
 ```
 
-## 8.7. Distributed Query
+## 8.7. SQL
+
+The SQL service provided by Hazelcast Node.js client allows you to query data stored in `IMap` declaratively.
+
+> **WARNING: The SQL feature is currently in beta. The compatibility between versions is not guaranteed. API might change between versions without notice. While in beta, SQL feature is tested against the same version of the IMDG, e.g 4.2.x
+> client is tested against 4.2.x IMDG server.**
+
+### Example: How to Query an IMap using SQL
+
+This SQL query returns map entries whose values are more than 1:
+
+```javascript
+const map = await client.getMap('my-distributed-map');
+await map.put('key1', 1);
+await map.put('key2', 2);
+await map.put('key3', 3);
+
+const result = client.getSqlService().execute(`SELECT __key, this FROM my-distributed-map WHERE this > 1`);
+
+for await(const row of result){
+    console.log(row); // {__key: 'key3', this: 3} and {__key: 'key2', this: 2}
+}
+```
+
+### Casting
```

Comment:
We have a separate section for casting, I think it will be better if we move this to there

Reply:
move what?

Reply:
I think now I got it, move this section to 8.7.3